File: src/org/j92/date/generator/date/CurrentDateGenerator.java
package org.j92.date.generator.date;

import com.intellij.ide.util.PropertiesComponent;
import com.intellij.openapi.project.Project;
import org.j92.date.generator.Settings;

import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;

public class CurrentDateGenerator implements DateGenerator {
    Project project;

    public CurrentDateGenerator(Project project) {
        this.project = project;
    }

    @Override
    public String generate() {
        String format = Settings.getInstance(this.project).format;
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
        return formatter.format(ZonedDateTime.now());
    }
}

File: src/org/j92/date/generator/date/CurrentDateGenerator.kt
package org.j92.date.generator.date

import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter

class CurrentDateGenerator(val pattern: String = "yyyy-MM-dd'T'HH:mm:ssXXX") : DateGenerator {
    override fun generate(zonedDateTime: ZonedDateTime): String {
        val formatter = DateTimeFormatter.ofPattern(pattern)
        return formatter.format(zonedDateTime)
    }
}
